One Factor Experiment : Isme basically ham Treatment check krte h ye sachme koi standout ho rha h ya bs random variability h

Scenario :
we have 20 days session wise return : Asin London and NY

In [2]:
import numpy as np 

In [3]:
np.random.seed(42)
Asian = np.random.uniform(0,3,10)
London = np.random.uniform(1,5,10)
NY = np.random.uniform(1,3,10)

In [4]:
xa = np.mean(Asian)
xl = np.mean(London)
xn = np.mean(NY)
a=np.concat((Asian,London,NY))
x = np.mean(a)

Variable between the Sessions

In [5]:
Vb = 10*(xa-x)**2 + 10*(xl-x)**2 + 10*(xn-x)**2

Variable within the session 

In [6]:
Va = np.sum((Asian - xa)**2)
Vl = np.sum((London - xl)**2)
Vn = np.sum((NY - xn)**2)
Vw = np.sum((Va,Vl,Vn))

Variable with total 

In [7]:
V = np.sum((a - x)**2)

H0 : There is no difference in the return of session so Alpha is 0 
because alpha is difference of sessions 

In [8]:
sb = Vb/2
sw = Vw/27
df1,df2 = 2,27

In [9]:
from scipy import stats
F = sb/sw
p = 1 - stats.f.cdf(F,df1,df2)
p

np.float64(0.05174292787770729)

As we took critical = 0.05 

0.0517 > 0.05 , p calc > p critical so, Fails to reject Null hypothesis. Hence, All session mean are equal

# Through Python Library

In [12]:
from scipy.stats import f_oneway
fstat,pvalue = f_oneway(Asian,London,NY)
print(pvalue,fstat)

0.051742927877707436 3.3114070918303393


Manually and through python Library also we got the same so our function is totally correct 

-------------------------------------------------##############################################----------------------------